### PR TITLE
Support chunks in ForwardDiff backend

### PR DESCRIPTION
**Pull request by devmotion**
I noticed that the ForwardDiff backend doesn't support custom chunk sizes. This PR tries to fix this issue.

### CODE REVIEW

File: src/forwarddiff.jl [hunk 1]
```diff
@@ -1,6 +1,9 @@
 using .ForwardDiff: ForwardDiff, DiffResults, StaticArrays
 
-struct ForwardDiffBackend <: AbstractBackend end
+struct ForwardDiffBackend{CS} <: AbstractBackend
+    ForwardDiffBackend{CS}() where {CS} = new{CS}()
```
Comment (mohamed82008) [inline src/forwarddiff.jl:4]: is this inner constructor necessary?

File: src/forwarddiff.jl [hunk 2]
```diff
@@ -1,6 +1,9 @@
...
+end
+ForwardDiffBackend(; chunksize=nothing) = ForwardDiffBackend{getchunksize(chunksize)}()
```
Comment (mohamed82008) [inline src/forwarddiff.jl:6]: perhaps put a type annotation on `chunksize`?

File: Project.toml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 name = "AbstractDifferentiation"
 uuid = "c29ec348-61ec-40c8-8164-b8c60e9d9f3d"
 authors = ["Mohamed Tarek <mohamed82008@gmail.com> and contributors"]
-version = "0.2.1"
+version = "0.3.0"
```
Comment (sethaxen) [inline Project.toml:4]: (how) is this a breaking change?

Comment (devmotion) [reply Project.toml:4]: The additional type parameter breaks uses of the `ForwardDiffBackend` type, e.g., `SomeOtherType{ForwardDiffBackend}` won't dispatch correctly anymore.

File: src/forwarddiff.jl (hunk 2 above)
Comment (devmotion) [reply src/forwarddiff.jl]: Done.

File: src/forwarddiff.jl (hunk 1 above)
Comment (devmotion) [reply src/forwarddiff.jl]: No :smile: I removed it.

File: src/forwarddiff.jl [hunk 4]
```diff
@@ -1,6 +1,9 @@
 using .ForwardDiff: ForwardDiff, DiffResults, StaticArrays
 
-struct ForwardDiffBackend <: AbstractBackend end
```
[the remark this answers is not in the document]
Comment (devmotion) [reply src/forwarddiff.jl:3]: I added some docstrings.

**Review by sethaxen — APPROVED** [review]

**Pull request merged**